internal/test/testdata/fake_func-e/fake_func-e.go

```diff
@@ -0,0 +1,39 @@
+// Copyright 2021 Tetrate
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This file needs to be here since we need to consume some of func-e project packages. When
+// building this program, we need to run the "go build" from within the func-e root project
+// directory (i.e. the place where the func-e's go.mod resides).
+
+package main
+
+import (
+	"context"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/tetratelabs/func-e/internal/moreos"
+	"github.com/tetratelabs/func-e/internal/version"
+)
+
+func main() {
+	envoy := filepath.Join(os.Getenv("FUNC_E_HOME"), "versions", string(version.LastKnownEnvoy), "envoy")
```

Comment:
this source doesn't exercise enough of how the subprocess is controlled, though I appreciate the start first. Also, we should be eagle eye on any dev (ex. envoy. here). Finally, the main should be runnable normally.. one way to test is to use an actual path to envoy in it.. you'd then possibly notice the paths aren't right here. I made a more complete example here, and hopefully by the time we are done there is less copy/pasta involved https://github.com/tetratelabs/func-e/compare/fake-deux